### PR TITLE
Text editor

### PR DESCRIPTION
**Pull request by vacho87**

### CODE REVIEW

File: TextEditor/TextEditor.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                Output += buffer;
+                Buffer = null;
+            }
+
+        }
+
+
+        public void Undo()
+        {
+            if (outputToUndo == null)
```
Comment (alex-al-skiba) [inline TextEditor/TextEditor.cs]: the most tricky thing is to make not only last change, but all the changes undone one by one. please implement it (will require redesign, please don't store all previous revisions of the output)

File: TextEditor/TextEditor.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace TextEditor
+{
+    class TextEditor
```
Comment (alex-al-skiba) [inline TextEditor/TextEditor.cs:7]: Please implement a set of unit tests for this class.

Comment (alex-al-skiba) [inline TextEditor/TextEditor.cs]: Please implement a set of unit tests for this class.

**Review by alex-al-skiba — CHANGES_REQUESTED** [review]

File: TextEditor/TextEditor.cs (hunk 1 above)
Comment (vacho87) [reply TextEditor/TextEditor.cs]: Suggestion applied

File: Exercises.sln [hunk 3]
```diff
@@ -9,6 +9,10 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Exercises.Tests", "Exercise
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "LINQ.Exercises", "LINQ.Exercises\LINQ.Exercises.csproj", "{A3B60DD0-BC38-4023-B36C-A1A843361AA5}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "TextEditor", "TextEditor\TextEditor.csproj", "{1B239D97-BB3E-4656-99F3-841581A9A7F2}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TextEditorTests", "TextEditorTests\TextEditorTests.csproj", "{D71DF171-8FD7-47D4-8525-6F4E5587CB91}"
```
Comment (alex-al-skiba) [inline Exercises.sln:14]: this project wasn't added to git. please add it.

File: TextEditor/ModificationLogs/DeletionLog.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        private readonly char deletedSymbol;
+        public DeletionLog(char delSymb)
+        {
+            deletedSymbol = delSymb;
+        }
+
+        public void UndoModification()
+        {
+            TextEditor.Output += deletedSymbol;
```
Comment (alex-al-skiba) [inline TextEditor/ModificationLogs/DeletionLog.cs]: Using public static properties, and especially modifying them is a very bad practice. You can avoid it by injecting an instance of TextEditor into DeletionLog (as a constructor parameter).

File: TextEditor/ModificationLogs/Paste_InsertionLog.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    public class Paste_InsertionLog : IModification
+    {
+        readonly int insertionLength;
+        public Paste_InsertionLog(int insertedTextLength)
+        {
+            insertionLength = insertedTextLength;
+        }
+        public void UndoModification()
+        {
+            TextEditor.Output = TextEditor.Output.Remove(TextEditor.Output.Length - insertionLength);
```
Comment (alex-al-skiba) [inline TextEditor/ModificationLogs/Paste_InsertionLog.cs]: same as with DeletionLog, don't work with static properties.

**Review by alex-al-skiba — CHANGES_REQUESTED** [review]

**Comment by alex-al-skiba** [comment]
Let's move forward with web API and eventually the BusinessTrip project

**Pull request merged**